core/src/consensus/tendermint/engine.rs

```diff
@@ -273,7 +267,7 @@ impl ConsensusEngine for Tendermint {
 
         let inner = self.inner.clone();
         let extension = service.register_extension(move |api| TendermintExtension::new(inner, timeouts, api));
-        let client = Weak::clone(self.client.read().as_ref().unwrap());
+        let client = Arc::downgrade(&self.client().unwrap());
```

Comment:
Note that this patch will changes the behavior (by small change)
The previous one panics only when the `self.client == None`, however, the changed one will panics 1. `self.client == None`, 2. `Weak::upgrade(self.client.unwrap()).is_none()`

Reply:
Well, it does change the behavior, but I don't think it matters.
`register_network_extension_to_service` is called only once at the initialization, and I think there's no possibility of panic in that situation.